### PR TITLE
refact(llm): separate user prompt configs

### PR DESCRIPTION
**Pull request by returnToInnocence**
Split settings by type

### CODE REVIEW

**Comment by ChenZiHong-Gavin** [comment]
hugegraph-llm/src/hugegraph_llm/config/config_prompt.yaml needs a liscense header

**Comment by returnToInnocence** [comment]
> hugegraph-llm/src/hugegraph_llm/config/config_prompt.yaml needs a liscense header

Roger that! I'll fix it right away!

**Comment by returnToInnocence** [comment]
> hugegraph-llm/src/hugegraph_llm/config/config_prompt.yaml needs a liscense header



> liscense header

Finished adding the apache liscense header, but we haven't split the data and operation yet. I plan to continue splitting later after lunch🫡

File: hugegraph-llm/src/hugegraph_llm/config/config.py [hunk 1]
```diff
@@ -125,3 +85,46 @@ def read_dotenv() -> dict[str, Optional[str]]:
...
+docs_build_rag: |
+{indented_docs_build_rag}
+        """
+        with open(yaml_file_path, 'w') as file:
+            file.write(yaml_content)
+        print(f"YAML file '{yaml_file_path}' updated successfully.")
+
+    def update_yaml_file(self):
+        print(f"Updating '{yaml_file_path}' with the latest attributes.")
+        self.save_to_yaml()
```
Comment (ChenZiHong-Gavin) [inline hugegraph-llm/src/hugegraph_llm/config/config.py:130]: a new line is needed

File: hugegraph-llm/src/hugegraph_llm/config/config_prompt.yaml [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        - Specify the name of the HugeGraph graph instance, it will automatically get the schema from it.
+    - Info extract head: The head of prompt of info extracting.
+    - Build mode: 
+        - Test Mode: Only extract vertices and edges from the file into memory (without building the vector index or 
+        writing data into HugeGraph)
+        - Import Mode: Extract the data and append it to HugeGraph & the vector index (without clearing any existing data)
+        - Clear and Import: Clear all existed RAG data(vector + graph), then rebuild them from the current input
+        - Rebuild Vector: Only rebuild vector index. (keep the graph data intact)
+        
+
```
Comment (ChenZiHong-Gavin) [inline hugegraph-llm/src/hugegraph_llm/config/config_prompt.yaml:118]: again, a new line is needed

File: hugegraph-llm/src/hugegraph_llm/config/config.py [hunk 3]
```diff
@@ -17,64 +17,24 @@
...
 
 from hugegraph_llm.utils.log import log
+from hugegraph_llm.config.config_data import ConfigData, PromptData
 
 dirname = os.path.dirname
 package_path = dirname(dirname(dirname(dirname(os.path.abspath(__file__)))))
 env_path = os.path.join(package_path, ".env")
+yaml_file_path = os.path.join(package_path, "src/hugegraph_llm/config/config_prompt.yaml")
 
+# TODO: We need to tidy up the partition settings
```
Comment (ChenZiHong-Gavin) [inline hugegraph-llm/src/hugegraph_llm/config/config.py:34]: remove this TODO if it is down?

File: hugegraph-llm/src/hugegraph_llm/demo/rag_web_demo.py [hunk 4]
```diff
@@ -542,15 +509,15 @@ def reranker_settings(reranker_type):
         gr.Markdown("""## 2. RAG with HugeGraph 📖""")
         with gr.Row():
             with gr.Column(scale=2):
-                inp = gr.Textbox(value="Tell me about Sarah.", label="Question", show_copy_button=True)
+                inp = gr.Textbox(value=settings.question, label="Question", show_copy_button=True)
```
Comment (ChenZiHong-Gavin) [inline hugegraph-llm/src/hugegraph_llm/demo/rag_web_demo.py:512]: Perhaps we need to standardize the input format here?
If `lines=2` is set to `answer_prompt_input`, `question` also needs it?

File: hugegraph-llm/src/hugegraph_llm/config/config.py (hunk 3 above)
Comment (returnToInnocence) [reply hugegraph-llm/src/hugegraph_llm/config/config.py:34]: get it!

File: hugegraph-llm/src/hugegraph_llm/demo/rag_web_demo.py (hunk 4 above)
Comment (returnToInnocence) [reply hugegraph-llm/src/hugegraph_llm/demo/rag_web_demo.py:512]: I agree. The user's question should also be able to support multi-line input, so I will make a simple change🫡

File: hugegraph-llm/src/hugegraph_llm/config/config_data.py [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+### Content Rule
+Please read the provided text carefully and identify any information that corresponds to the vertices and edges defined in the schema. For each piece of information that matches a vertex or edge, format it according to the following JSON structures:
+#### Vertex Format:
+{"id":"vertexLabelID:entityName","label":"vertexLabel","type":"vertex","properties":{"propertyName":"propertyValue", ...}}
+#### Edge Format:
+{"label":"edgeLabel","type":"edge","outV":"sourceVertexId","outVLabel":"sourceVertexLabel","inV":"targetVertexId","inVLabel":"targetVertexLabel","properties":{"propertyName":"propertyValue",...}}
+Also follow the rules: 
+1. Don't extract property fields that do not exist in the given schema
+2. Ensure the extracted property is in the same type as the schema (like 'age' should be a number)
+3. If there are multiple primary keys, the strategy for generating VID is: vertexlabelID:pk1!pk2!pk3 (pk means primary key, and '!' is the separator)
```
Comment (imbajin) [inline hugegraph-llm/src/hugegraph_llm/config/config_data.py:98]: read the config/str from `config_prompt.yaml` directly?

(seems no need to store them again)

Comment (returnToInnocence) [reply hugegraph-llm/src/hugegraph_llm/config/config_data.py:98]: get it🫡

Comment (returnToInnocence) [reply hugegraph-llm/src/hugegraph_llm/config/config_data.py:98]: get it

**Review by imbajin — APPROVED** [review]

**Pull request merged**